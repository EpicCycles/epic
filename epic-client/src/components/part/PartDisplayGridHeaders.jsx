import React from "react";
import PartHeaders from "../part/PartHeaders";
import * as PropTypes from "prop-types";
import SupplierProductFieldHeaders from "../supplierProduct/SupplierProductFieldHeaders";
import {fixedHeaderClassname} from "../app/model/helpers/display";

const PartDisplayGridHeaders = (props) => {
    return <div className="grid-row grid-row--header " key="part-display-grid-header-row">
        <div
            className={`grid-item--header ${props.className} ${fixedHeaderClassname(props.lockFirstColumn)}`}
            key="part-display-grid-header-section"
        >
            Section
        </div>
        <PartHeaders
            showErrors={props.showErrors}
            className={props.className}
            key="part-display-grid-header-part-headers"
        />
        {props.showSupplierProducts && <SupplierProductFieldHeaders
            className={props.className}
            showErrors={props.showErrors}
            key="part-display-grid-header-sp-headers"
        />}
        {props.includeActions && <div
            className={`grid-item--header ${props.className}`}
            key="part-display-grid-header-actions"
        >
            Actions
        </div>}
    </div>;
};
PartDisplayGridHeaders.defaultProps = {
    className: "",
};
PartDisplayGridHeaders.propTypes = {
    lockFirstColumn: PropTypes.bool,
    showSupplierProducts: PropTypes.bool,
    showErrors: PropTypes.bool,
    includeActions: PropTypes.any,
    className: PropTypes.string,
};
export default PartDisplayGridHeaders;