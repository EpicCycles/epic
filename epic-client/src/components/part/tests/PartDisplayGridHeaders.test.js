import React from 'react';
import PartDisplayGridHeaders from "../PartDisplayGridHeaders";

test('should display just part headers when supplier products no requested', () => {
    const component = shallow(<PartDisplayGridHeaders/>);
    expect(component).toMatchSnapshot();
});
test('should display all headers with locking and an extra class', () => {
    const component = shallow(<PartDisplayGridHeaders
        lockFirstColumn={true}
        showSupplierProducts={true}
        showErrors={true}
        className={"pink"}
    />);
    expect(component).toMatchSnapshot();
});
test('should display all headers with locking and an extra class and actions', () => {
    const component = shallow(<PartDisplayGridHeaders
        lockFirstColumn={true}
        showSupplierProducts={true}
        showErrors={true}
        className={"pink"}
        includeActions={true}
    />);
    expect(component).toMatchSnapshot();
});