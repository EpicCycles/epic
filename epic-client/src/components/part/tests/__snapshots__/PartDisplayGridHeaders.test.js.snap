// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display all headers with locking and an extra class 1`] = `
<div
  className="grid-row grid-row--header "
>
  <div
    className="grid-item--header pink grid-header--fixed-left"
  >
    Section
  </div>
  <PartHeaders
    className="pink"
    showErrors={true}
  />
  <SupplierProductFieldHeaders
    className="pink"
    lockFirstColumn={false}
    showErrors={true}
  />
</div>
`;

exports[`should display just part headers when supplier products no requested 1`] = `
<div
  className="grid-row grid-row--header "
>
  <div
    className="grid-item--header  "
  >
    Section
  </div>
  <PartHeaders
    className=""
    showErrors={false}
  />
</div>
`;
